File: data/models/Playlist.js
// models/Playlist.js
const mongoose = require( 'mongoose' );
const BaseSchema = require( './BaseSchema' );

const PlaylistSchema = new mongoose.Schema( {
  user:    { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name:    { type: String, required: true },
  request: { type: mongoose.Schema.Types.ObjectId, ref: 'Request' },
  link:    { type: String },
} );

PlaylistSchema.add( BaseSchema );

module.exports = mongoose.model( 'Playlist', PlaylistSchema );

File: backend/src/models/Playlist.py
# Playlist.py

from mongoengine import StringField, ReferenceField
from models.BaseSchema import BaseSchema

class Playlist(BaseSchema):
    user    = ReferenceField('users', required=True)
    name    = StringField(required=True)
    request = ReferenceField('requests', null=True)
    link    = StringField(required=True)

    meta = {
        'collection': 'playlists'  # Optional: specify the collection name
    }
